File: json_to_flutter/models/stack.dart
import 'index.dart';

class Stack extends Widget {
  Stack(double x, double y, double w, double h, double gw, double gh,
      {String id = "-1"})
      : super(x, y, w, h, gw, gh, id);

  @override
  String generateWidget(No node) {
    List<String> widgets = [];
    node.children?.forEach((f) {
      widgets.add("${f.widget.generateWidget(f)}");
    });
    widgets = widgets.reversed.toList();
    String wid = "$widgets";
    return "Stack(children:<Widget>$wid,)";
  }
}

File: src/app/jsonToFlutter/models/stack.js
class Stack {
  constructor(x, y, w, h, gw, gh) {
    this.type = Stack;
    this.id = "-1";
    this.x = x;
    this.y = y;
    this.w = w;
    this.h = h;
    this.gw = gw;
    this.gh = gh;
  }

  generateWidget(no) {
    let widgets = [];
    if (no.children != null) {
      no.children.forEach(function (f) {
        widgets.push(`${f.widget.generateWidget(f)}`);
      });
    }
    widgets.reverse();
    let wid = `${widgets}`;
    return `Stack(alignment: Alignment.center,children:<Widget>[${wid}],)`;
  }
}

module.exports = { Stack };